components/zcash_address/src/encoding.rs

```diff
@@ -0,0 +1,298 @@
+                },
+                data,
+            ),
+            AddressKind::Unified(data) => encode_bech32m(
+                match self.net {
+                    Network::Main => unified::MAINNET,
+                    Network::Test => unified::TESTNET,
+                    Network::Regtest => unified::REGTEST,
+                },
+                &data.to_bytes(),
+            ),
+            AddressKind::P2pkh(data) => encode_b58(
+                if let Network::Main = self.net {
+                    p2pkh::MAINNET
+                } else {
+                    p2pkh::TESTNET
+                },
+                data,
+            ),
+            AddressKind::P2sh(data) => encode_b58(
+                if let Network::Main = self.net {
+                    p2sh::MAINNET
+                } else {
+                    p2sh::TESTNET
+                },
+                data,
+            ),
+        };
+        write!(f, "{}", encoded)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::kind::unified;
+
+    fn encoding(encoded: &str, decoded: ZcashAddress) {
+        assert_eq!(decoded.to_string(), encoded);
+        assert_eq!(encoded.parse(), Ok(decoded));
+    }
+
+    #[test]
+    fn sprout() {
+        encoding(
+            "zc8E5gYid86n4bo2Usdq1cpr7PpfoJGzttwBHEEgGhGkLUg7SPPVFNB2AkRFXZ7usfphup5426dt1buMmY3fkYeRrQGLa8y",
+            ZcashAddress { net: Network::Main, kind: AddressKind::Sprout([0; 64]) },
+        );
+        encoding(
+            "ztJ1EWLKcGwF2S4NA17pAJVdco8Sdkz4AQPxt1cLTEfNuyNswJJc2BbBqYrsRZsp31xbVZwhF7c7a2L9jsF3p3ZwRWpqqyS",
+            ZcashAddress { net: Network::Test, kind: AddressKind::Sprout([0; 64]) },
+        );
+    }
+
+    #[test]
+    fn sapling() {
+        encoding(
+            "zs1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqpq6d8g",
+            ZcashAddress {
+                net: Network::Main,
+                kind: AddressKind::Sapling([0; 43]),
+            },
+        );
+        encoding(
+            "ztestsapling1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqfhgwqu",
+            ZcashAddress {
+                net: Network::Test,
+                kind: AddressKind::Sapling([0; 43]),
+            },
+        );
+        encoding(
+            "zregtestsapling1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqknpr3m",
+            ZcashAddress {
+                net: Network::Regtest,
+                kind: AddressKind::Sapling([0; 43]),
+            },
+        );
+    }
+
+    #[test]
+    fn unified() {
+        encoding(
+            "u1cd8yzk5mdn4n9r8c24tp8j8e9ethw3rr7ker5zhew3kycyyxggdzfkcq5f9yf2jv8m5ar8krncsntlfpx3p4azvwrkp8z74t3vu4kqq2",
+            ZcashAddress {
+                net: Network::Main,
+                kind: AddressKind::Unified(unified::Address(vec![unified::Receiver::Sapling(
+                    [0; 43],
+                )])),
```

Comment:
Something is odd about the formatting of this being different from lines 233 and 240; was rustfmt re-run?

Reply:
It appears that `rustfmt` (either run directly or via `cargo fmt`, or even `cargo fmt --all`) is not touching this code at all. I've seen this before for test code, and I don't know why. Maybe https://github.com/rust-lang/rustfmt/issues/3253 ? I didn't think that applied to tests.